v3/src/transports/http/handler.ts

```diff
@@ -0,0 +1,145 @@
+import { GraphQLSchema } from "graphql";
+import { GraphQLRequest } from "../../types";
+
+/**
+ * A factory function that receives an instance of `ApolloServer` and returns
+ * a `RequestHandler` that can be used with Node.js' `http.createServer`, or
+ * Express' `app.use`.
+ *
+ * @param apollo An instance of `ApolloServer`
+ */
+export const httpHandler: (
+  schema: GraphQLSchema,
+) => RequestListener = schema => {
+  if (!(schema instanceof GraphQLSchema)) {
+    throw new Error("Must pass a schema.");
+  }
+
+  /**
+   * Returns the handler that can be passed to the HTTP framework that
+   * respects the `(req, res)` pattern (e.g. Express or Node.js).
+   */
+  return async function httpRequestListener(req, res): Promise<void> {
+    if (!req) {
+      return responseAsInternalServerError(
+        res,
+        "Missing request on HTTP request handler invocation.",
+      );
+    }
+
+    if (!res) {
+      return responseAsInternalServerError(
+        res,
+        "Missing response sink on HTTP request handler invocation.",
+      );
+    }
+
+    let parsedRequest: GraphQLRequest;
+    try {
+      /**
+       * TODO: Need to assert at runtime that the properties we expect to
+       * be there are present.
+       */
+      parsedRequest = await jsonBodyParse(req);
+    } catch (err) {
+      // TODO(AS3) In order to limit error codes to a single place, this may
+      // be well-served to be a `GraphQLError`.
+      return responseAsInternalServerError(res, "Error parsing body");
+    }
+
+    /**
+     * Maps the incoming request to the shape that the Apollo HTTP transport
+     * expects it to be in.
+     */
+    const httpGraphqlRequest: IHttpRequest = {
+      parsedRequest,
+      url: req.url,
+      headers: req.headers,
+      // The `method` property, while optional in `http.IncomingMessage` type,
+      // is guaranteed to be present on extensions of `http.Server` instances.
+      // Ref: https://git.io/JeM4V
+      method: req.method!,
+    };
+
+    const httpGraphqlResponse = await processHttpRequest({
+      schema,
+      request: httpGraphqlRequest,
+    });
+
+    // Map headers
+    for (const [key, value] of Object.entries(httpGraphqlResponse.headers)) {
+      res.setHeader(key, value || '');
+    }
+
+    // Map status codes
+    res.statusCode = httpGraphqlResponse.statusCode;
+    res.statusMessage = httpGraphqlResponse.statusMessage || '';
+
+    // Map bodies
+    //
+    // Using `Readable.from(body)` would be great, and would allow a direct
+    // pipe, but that's not supported until Node.js 12. Something from
+    // `readable-stream` might be a suitable alternative.
+    // https://www.npmjs.com/package/readable-stream
+    for await (const chunkedHttpResponse of httpGraphqlResponse.body) {
+      res.write(JSON.stringify(chunkedHttpResponse) + '\n\n');
+    }
+    res.end();
+  };
+};
+
+/**
+ * Called in the event of a critical error within the HTTP handler.
+ *
+ * @param res Called
```

Comment:
Well, this is certainly not a complete sentence.
```suggestion
 * @param res 
```